tools/src/h5diff/ph5diff_main.c

```diff
@@ -165,34 +166,44 @@ ph5diff_worker(int nID)
             diffs.nfound  = diff(file1_id, args.name1, file2_id, args.name2, &(args.opts), &(args.argdata));
             diffs.not_cmp = args.opts.not_cmp;
 
-            /* If print buffer has something in it, request print token.*/
-            if (outBuffOffset > 0) {
+            if ((outBuffOffset == 0) && !overflow_file)
+                /* Nothing to print. Send diffs to manager */
+                MPI_Send(&diffs, sizeof(diffs), MPI_BYTE, 0, MPI_TAG_DONE, MPI_COMM_WORLD);
+            else {
+                /*
+                 * If print buffer or overflow file have something in
+                 * them, request print token.
+                 */
                 MPI_Send(NULL, 0, MPI_BYTE, 0, MPI_TAG_TOK_REQUEST, MPI_COMM_WORLD);
 
                 /* Wait for print token. */
                 MPI_Recv(NULL, 0, MPI_BYTE, 0, MPI_TAG_PRINT_TOK, MPI_COMM_WORLD, &Status);
 
-                /* When get token, send all of our output to the manager task and then return the token */
-                for (i = 0; i < outBuffOffset; i += PRINT_DATA_MAX_SIZE)
-                    MPI_Send(outBuff + i, PRINT_DATA_MAX_SIZE, MPI_CHAR, 0, MPI_TAG_PRINT_DATA,
-                             MPI_COMM_WORLD);
+                if (outBuffOffset > 0) {
+                    /* When get token, send all of our output to the manager task and then return the token */
+                    for (i = 0; i < outBuffOffset; i += PRINT_DATA_MAX_SIZE)
+                        MPI_Send(outBuff + i, PRINT_DATA_MAX_SIZE, MPI_CHAR, 0, MPI_TAG_PRINT_DATA,
+                                 MPI_COMM_WORLD);
+                }
 
-                /* An overflow file exists, so we send it's output to the manager too and then delete it */
+                /* An overflow file exists, so we send its output to
+                 * the manager too and then delete it.
+                 */
                 if (overflow_file) {
-                    char out_data[PRINT_DATA_MAX_SIZE];
+                    char out_data[PRINT_DATA_MAX_SIZE + 1];
```

Comment:
Not strictly necessary, but matches some other places and reduces future chances of accidentally using a non-NUL-terminated buffer